eclass/rebar3.eclass

```diff
@@ -0,0 +1,258 @@
+# Copyright 1999-2016 Gentoo Foundation
```

Comment:
why is it here?

Reply:
How is it relevant to the ebuild update? It should not be here, make it atomic

Reply:
see here about commits policy https://www.gentoo.org/glep/glep-0066.html

Reply:
and also please always keep your changes squashed

Reply:
What section of the commit policy should I be looking at?

This is not an ebuild update, it's a new package - modelled after the existing "dev-util/rebar" which has been deprecated upstream, but it's still needed by many Erlang packages.

The new rebar3.eclass - modelled after the existing rebar.eclass - is needed for Erlang packages depending on rebar3. It's all rather straight-forward.

As for squashing changes at my end, why bother? You're not merging pull requests using GitHub, so you just need a diff like https://patch-diff.githubusercontent.com/raw/gentoo/gentoo/pull/9360.patch to make your own commit to the main tree, right?

Reply:
>As for squashing changes at my end, why bother? You're not merging pull requests using GitHub, so you just need a diff like https://patch-diff.githubusercontent.com/raw/gentoo/gentoo/pull/9360.patch to make your own commit to 

We do not merge bunch of patches. The url you provided picks the first commit, not the changes you made afterwards. If we merged bunch of them it would create lots of needless commits.

Reply:
The new eclass depends on this new package and all other packages using this new build tool are supposed to do it from this eclass. The two are a "logical unit", to use the terminology of GLEP 66.

Reply:
> The url you provided picks the first commit, not the changes you made afterwards.

It picks all of them, keep scrolling down. The formatting looks like a concatenation of individual diff files obtained from the respective Git commits.

> If we merged bunch of them it would create lots of needless commits.

How exactly do you merge these pull requests?

Reply:
>It picks all of them, keep scrolling down. The formatting looks like a concatenation of individual diff files obtained from the respective Git commits.

It adds useless complexity for minor changes as well.

>The new eclass depends on this new package and all other packages using this new build tool are supposed to do it from this eclass. The two are a "logical unit", to use the terminology of GLEP 66.

1.) it is still irrelevant an should be at least provided as a separate atomic commit
2.) new eclasses can not be introduced without prior discussion in the gentoo-dev mailing lists
3.) your eclas seems to be just copy-pasted from the `rebar.eclass` and has an invalid entries from te start (I did not diff them though), `@ECLASS`, `@MAINTAINER` and `@AUTHOR` do not reflect reality

Reply:
> new eclasses can not be introduced without prior discussion in the gentoo-dev mailing lists

It's similar enough to an existing eclass.

> your eclas seems to be just copy-pasted from the rebar.eclass

It is, with the necessary changes to make it use the "rebar3" binary.

> @ECLASS, @MAINTAINER and @AUTHOR do not reflect reality

I updated the first two variables. The third one can remain unchanged.